cmd/node/main.go

```diff
@@ -2254,7 +2284,7 @@ func createNode(
 		return nil, errors.New("error creating node: " + err.Error())
 	}
 
-	err = nd.StartHeartbeat(config.Heartbeat, version, preferencesConfig.Preferences)
+	err = nd.StartHeartbeat(*config, version, preferencesConfig.Preferences)
```

Comment:
It would have been much easier in (code which would have been changed in: hearbeatHandler + heartbeatHandler_test) if you have been kept the config.HeartBeat and used an additional bool parameter here like fullArchive -> config.StoragePruning.FullArchive. Anyway, you can leave it like this now.